pinot-segment-local/src/test/java/org/apache/pinot/segment/local/segment/index/loader/SegmentPreProcessorTest.java

```diff
@@ -1115,6 +1116,18 @@ private void testIfNeedProcess()
       assertFalse(processor.needProcess());
     }
 
+    // No preprocessing needed if required to add certain index on non-existing, sorted or non-dictionary column.
+    for (Consumer<IndexLoadingConfig> prepFunc : createConfigPrepFunctionNeedNoops()) {
```

Comment:
The purpose of a test is to prevent someone from breaking the implementation unexpectedly in the future as much as it is to demonstrate that a change/feature works, so a test must surface diagnostics about failure. This test was already problematic in this regard, but this change doesn't make it better: If this test ever fails, someone will need to put a breakpoint in this loop to find out when the test actually fails, and when they do that, they will see a `Consumer` without any identifiable state with a class name like "...SegmentPreProcessorTest$$Lambda$86/0x00000008001a5c40". The only way to know which consumer the test fails for would be to modify the test to add a counter to be incremented on each iteration, and then look at the nth position in the list created in `createConfigPrepFunctionNeedNoops`. 

This can all be circumnavigated by using `DataProvider` to create combinations of test inputs (including a descriptive test name) to supply to generic testing logic.

Reply:
didn't use DataProvider but a simple Map to provide the test cases, as the test cases are checked both individually and used to load up a temp config upfront. But anyway, I added test name to assertions for easy debugging.